apis/v1alpha1/destination.go

```diff
@@ -18,132 +18,60 @@ package v1alpha1
 
 import (
 	"context"
-	"net/url"
-	"path"
-	"strings"
-
 	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/equality"
-
 	"knative.dev/pkg/apis"
 )
 
 // Destination represents a target of an invocation over HTTP.
 type Destination struct {
-	// ObjectReference points to an Addressable.
-	*corev1.ObjectReference `json:",inline"`
+	// Ref points to an Addressable.
+	// +optional
+	Ref *corev1.ObjectReference `json:"ref,omitempty"`
 
-	// URI is for direct URI Designations.
+	// URI can be a absolute URI which points to the URI designation or it can be a relative URI will be merged to the URI retrieved from Ref.
+	// +optional
 	URI *apis.URL `json:"uri,omitempty"`
-
-	// Path is used with the resulting URL from Addressable ObjectReference or URI. Must start
-	// with `/`. An empty path should be represented as the nil value, not `` or `/`.  Will be
-	// appended to the path of the resulting URL from the Addressable, or URI.
-	Path *string `json:"path,omitempty"`
 }
 
 // NewDestination constructs a Destination from an object reference as a convenience.
-func NewDestination(obj *corev1.ObjectReference, paths ...string) (*Destination, error) {
-	dest := &Destination{
-		ObjectReference: obj,
-	}
-	err := dest.AppendPath(paths...)
-	if err != nil {
-		return nil, err
-	}
-	return dest, nil
+func NewDestination(obj *corev1.ObjectReference) (*Destination, error) {
```

Comment:
I doubled checked it. At first, the function is the helper function for `TestDestinationWithPath`. Now since we delete `Path` and delete `TestDestinationWithPath`. I think we can delete it. I also checked repo of knative-gcp, I don't find any usage of function `NewDestination`. I think it's safe to remove it.